src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs

```diff
@@ -28,26 +28,20 @@ public class EnableSecureSourceCodeHashing : WindowsBinaryAndPdbSkimmerBase, IOp
         protected override IEnumerable<string> MessageResourceNames => new string[] {
             nameof(RuleResources.BA2004_Pass),
             nameof(RuleResources.BA2004_Warning_NativeWithInsecureStaticLibraryCompilands),
-            nameof(RuleResources.BA2004_Warning_Managed),
+            nameof(RuleResources.BA2004_Error_Managed),
             nameof(RuleResources.BA2004_Error_NativeWithInsecureDirectCompilands),
             nameof(RuleResources.NotApplicable_InvalidMetadata)
         };
 
         public override AnalysisApplicability CanAnalyzePE(PEBinary target, Sarif.PropertiesDictionary policy, out string reasonForNotAnalyzing)
         {
-            PE portableExecutable = target.PE;
-            AnalysisApplicability result = AnalysisApplicability.NotApplicableToSpecifiedTarget;
-
-            reasonForNotAnalyzing = MetadataConditions.ImageIsILOnlyAssembly;
-            if (portableExecutable.IsILOnly) { return result; }
-
-            // TODO: currently our test binary for this check is a dll that does not
-            // compile against any external library. BinSkim regards this as a resource
-            // only binary and skips the test. We should improve the IsResourceOnly
-            // helper to properly identify this dependency-free test binary as code.
+            Pdb di = target.Pdb;
 
-            //            reasonForNotAnalyzing = MetadataConditions.ImageIsResourceOnlyBinary;
-            //            if (portableExecutable.IsResourceOnly) { return result; }
+            if (target.PE.IsManaged && di == null)
+            {
+                reasonForNotAnalyzing = MetadataConditions.CouldNotLoadPdb;

```

Comment:
>reasonForNotAnalyzing = MetadataConditions.CouldNotLoadPdb [](start = 16, length = 58)

in other checks, the absence of a PDB results in an error level notification, not a 'not applicable' message. 'Not applicable' is not the right return value here, as this return value means 'the binary isn't a valid scan target'. that's not true here, managed code *is* a valid scan target, the problem is that we can't analyze it due to a missing pdb. you should go look and see how the native pdb reading errors handle a missing pdb.

Reply:
can you validate again? pushed new changes. Thank you

---
In reply to: [563214549](https://github.com/microsoft/binskim/pull/335#discussion_r563214549) [](ancestors = 563214549)